pulpcore/cli/common/openapi.py

```diff
@@ -164,54 +176,160 @@ def extract_params(
+
+    def validate_array(self, schema: Any, name: str, value: Any) -> List[Any]:
+        if not isinstance(value, List):
+            raise OpenAPIValidationError(_("'{name}' is expected to be a list.").format(name=name))
+        item_schema = schema["items"]
+        return [self.validate_schema(item_schema, name, item) for item in value]
+
+    def validate_string(self, schema: Any, name: str, value: Any) -> Union[str, UploadType]:
+        schema_format = schema.get("format")
+        if schema_format == "date":
+            if not isinstance(value, datetime.date):
+                raise OpenAPIValidationError(
+                    _("'{name}' is expected to be a date.").format(name=name)
+                )
+            return value.strftime(ISO_DATE_FORMAT)
+        elif schema_format == "date-time":
+            if not isinstance(value, datetime.datetime):
+                raise OpenAPIValidationError(
+                    _("'{name}' is expected to be a datetime.").format(name=name)
+                )
+            return value.strftime(ISO_DATETIME_FORMAT)
+        elif schema_format == "bytes":
+            if not isinstance(value, bytes):
+                raise OpenAPIValidationError(
+                    _("'{name}' is expected to be bytes.").format(name=name)
+                )
+            return base64.b64encode(value)
+        elif schema_format == "binary":
+            if not isinstance(value, (bytes, BufferedReader)):
+                raise OpenAPIValidationError(
+                    _("'{name}' is expected to be binary.").format(name=name)
+                )
+            return value
+        else:
+            if not isinstance(value, str):
+                raise OpenAPIValidationError(
+                    _("'{name}' is expected to be a string.").format(name=name)
+                )
+            return value
+
+    def validate_integer(self, schema: Any, name: str, value: Any) -> int:
+        if not isinstance(value, int):
+            raise OpenAPIValidationError(
+                _("'{name}' is expected to be an integer.").format(name=name)
+            )
+        minimum = schema.get("minimum")
+        if minimum is not None and value < minimum:
+            raise OpenAPIValidationError(
+                _("'{name}' is violating the minimum constraint").format(name=name)
+            )
+        maximum = schema.get("maximum")
+        if maximum is not None and value > maximum:
+            raise OpenAPIValidationError(
+                _("'{name}' is violating the maximum constraint").format(name=name)
+            )
+        return value
+
+    def validate_number(self, schema: Any, name: str, value: Any) -> float:
+        # https://swagger.io/specification/#data-types describes float and double.
+        # Python does not distinguish them.
+        if not isinstance(value, float):
+            raise OpenAPIValidationError(
+                _("'{name}' is expected to be an integer.").format(name=name)
```

Comment:
```suggestion
                _("'{name}' is expected to be an float.").format(name=name)
```

Reply:
Actually in the OpenAPI world they call it "number".